File: javascript/leetcode/TwoPointers/ValidPalindrome.js
// A phrase is a palindrome if, after converting all uppercase letters into lowercase letters and removing all non-alphanumeric characters, it reads the same forward and backward. Alphanumeric characters include letters and numbers.

// Given a string s, return true if it is a palindrome, or false otherwise.


const isPalindrome = function(s) {
    let stack = [];
    let sLength = s.length-1;
    for (let i = 0; i <= sLength; i++) {
        let x = s[i].toLowerCase();
        if (x.charCodeAt(0) >= 97 && x.charCodeAt(0) <= 122 || x.charCodeAt(0) >= 48 && x.charCodeAt(0) <= 57) {
            stack.push(x);
        }
    }
    let i = 0;
    let j = stack.length - 1;
    while (i < j) {
        if (stack[i] !== stack[j]) {
            return false;
        }
        i++;
        j--;
    }
    return true;
};

console.log(isPalindrome(".P"));

File: Python/leetcode/Roadmap/TwoPointer/valid_palindrome.py
# Given a string s, return true if it is a palindrome, otherwise return false.

# A palindrome is a string that reads the same forward and backward. It is also case-insensitive and ignores all non-alphanumeric characters.

class Test:
    def palindrome(self, s):
        str = ""
        for c in s:
            if c.isalpha() or c.isdigit():
                str += c.lower()

        i = 0
        j = len(str) - 1
        for i in range(0, len(str)):
            if str[i] != str[j]:
                return False
            i += 1
            j -= 1
        return True
    
t = Test()
print(t.palindrome("Racecar"))